sklearn/utils/estimator_checks.py

```diff
@@ -760,7 +763,10 @@ def _check_transformer(name, transformer_orig, X, y):
         # raises error on malformed input for transform
         if hasattr(X, 'T'):
             # If it's not an array, it does not have a 'T' property
-            assert_raises(ValueError, transformer.transform, X.T)
+            with assert_raises(ValueError, msg="The transformer {name} does "
```

Comment:
By malformed here do we mean "has a different number of features to when fitting"?